ArduCopter/mode_auto.cpp

```diff
@@ -117,6 +112,11 @@ void ModeAuto::run()
     }
 }
 
+bool ModeAuto::allows_arming(bool from_gcs) const
+{
+    return (copter.g2.auto_options & (int32_t)Options::AllowArming) != 0;
```

Comment:
```suggestion
    return (copter.g2.auto_options & (uint32_t)Options::AllowArming) != 0;
```

(storage type might be signed, but that's a problem with Param, and we want unsigned for a mask)

Reply:
I guess this is existing code not changed by my PR to I'll leave this suggestion out.

Reply:
I've fixed the equivalent for guided_options, thanks!

Reply:
actually I've fixed the auto mode one as well now and re-tested that arming within auto still works